homeassistant/components/sonos/__init__.py

```diff
@@ -31,6 +61,19 @@
 )
 
 
+class SonosData:
+    """Storage class for platform global data."""
+
+    def __init__(self):
+        """Initialize the data."""
+        self.discovered = {}
+        self.media_player_entities = {}
```

Comment:
We shouldn't store entities in a shared container, but this is existing code that has been relocated and this PR is already large so let's fix this in a follow up